glusterd2/volgen/common.go

```diff
@@ -0,0 +1,244 @@
+package volgen
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+
+	config "github.com/spf13/viper"
+)
+
+const (
+	thinArbiterOptionName  = "replicate.thin-arbiter"
+	thinArbiterDefaultPort = "24007"
```

Comment:
question: why this port is hardcoded? cant we have it configurable?

Reply:
We can change this to take glusterd2's port by default. I will send different PR for the same, since it is not part of volgen changes.

Reply:
can you create an issue for the same, so we wont miss this

Reply:
ok

Reply:
thin arbiter node is not part of gluster cluster. Any default value is not sufficient since glusterd2 will not be run in the thin arbiter node. We already have option to specify thin arbiter port while specifying the thin arbiter brick. I don't see much advantage of changing this to use glusterd2's port.